src/controller/java/templates/ClusterInfo-java.zapt

```diff
@@ -240,7 +240,7 @@ public class ClusterInfoMapping {
 
       {{/if}}
 
-      public static class Delegated{{asUpperCamelCase name}}AttributeCallback implements ChipClusters.{{asUpperCamelCase ../name}}Cluster.{{asUpperCamelCase name}}AttributeCallback, DelegatedClusterCallback {
+      public static class Delegated{{asUpperCamelCase parent.name}}Cluster{{asUpperCamelCase name}}AttributeCallback implements ChipClusters.{{asUpperCamelCase ../name}}Cluster.{{asUpperCamelCase name}}AttributeCallback, DelegatedClusterCallback {
```

Comment:
Sounds like this is done because `AttributeList` generates collisions ?

Reply:
Yes, exactly.  Maybe I should look into generating separately for global attributes.... I'll give that a stab.